### PR TITLE
Add IUT monitoring commands for ETR

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: eiffel-community/etos#24

### Description of the Change

Added the 'commands' preparation step for IUT providers.
Changed so that IUT monitoring starts within the workspace.
Added execution of a script in IUT monitoring.
Added a global artifacts directory.

### Alternate Designs

This could be designed and added to ESR instead, however I think that adding it to ETR which had the support is less work and it already have the infrastructure ready to do it.
A longer discussion on whether this shall be added to ESR instead needs to be had.

### Benefits

IUT Monitoring adds the ability to monitor the IUT during test execution and provide more information about the item under test.

### Possible Drawbacks

It's extra complexity, more things that can go wrong and a high learning curve for our users.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

File: src/etos_test_runner/lib/iut_monitoring.py [hunk 1]
```diff
@@ -14,25 +14,79 @@
...
+        :type output: filedescriptor
+        """
+        self.logger.info("Reading output from %r", output)
+        for line in iter(output.readline, b''):
+            self.logger.info(line.decode("utf-8"))
+        output.close()
+
     def start_monitoring(self):
         """Start monitoring IUT."""
+        scripts = self.config.get("scripts") or []
```
Comment (fredjn) [inline src/etos_test_runner/lib/iut_monitoring.py:58]: This assumes a proper shebang. Would it be wrong to only support for instance bash? or python? or is that to restrictive?

**Review by fredjn — CHANGES_REQUESTED** [review]
No tests? ;)

**Review by andjoe-axis — APPROVED** [review]
+1

File: src/etos_test_runner/lib/iut_monitoring.py (hunk 1 above)
Comment (t-persson) [reply src/etos_test_runner/lib/iut_monitoring.py:58]: I believe that it would be a bit too restrictive.
We cannot foresee what others want to do.

**Review by fredjn — APPROVED** [review]

**Pull request merged**